### PR TITLE
schedule and schedule category

### PR DESCRIPTION
**Pull request by mrezav**
Module : General
------------------------------

Master : Schedule Category
Create master schedule category, create also seeder for this table like : 'meeting','aoc', etc

Planning : Schedule
Create new menu in general call 'Planning', schedule have category and department relation, every relation use code as foreign key.

### CODE REVIEW

File: database/seeds/DatabaseSeeder.php [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 use Illuminate\Database\Seeder;
 
+use Modules\General\Database\seeds\ScheduleCategorySeederTableSeeder;
```
Comment (muhammadcahya) [inline database/seeds/DatabaseSeeder.php:5]: Ini cukup ScheduleCategorySeeder aja

File: modules/General/Database/seeds/ScheduleCategorySeederTableSeeder.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+     */
+    public function run()
+    {
+        $now = now();
+        $data = [
+            ['id'=>Str::uuid(), 'code' => 'Met', 'name' => 'Meeting', 'created_by' => 2, 'updated_by' => 2,'created_at' => $now, 'updated_at' => $now, 'is_active' => true],
+            ['id'=>Str::uuid(), 'code' => 'Int', 'name' => 'Interview', 'created_by' => 2, 'updated_by' => 2,'created_at' => $now, 'updated_at' => $now, 'is_active' => true],
+            ['id'=>Str::uuid(), 'code' => 'Din', 'name' => 'Dinner', 'created_by' => 2, 'updated_by' => 2,'created_at' => $now, 'updated_at' => $now, 'is_active' => true],
+            ['id'=>Str::uuid(), 'code' => 'Aoc', 'name' => 'Aoc', 'created_by' => 2, 'updated_by' => 2,'created_at' => $now, 'updated_at' => $now, 'is_active' => true],
+            ['id'=>Str::uuid(), 'code' => 'Vis', 'name' => 'Visit', 'created_by' => 2, 'updated_by' => 2,'created_at' => $now, 'updated_at' => $now, 'is_active' => true],
```
Comment (muhammadcahya) [inline modules/General/Database/seeds/ScheduleCategorySeederTableSeeder.php:24]: - Semua code klo ngga besar semua kecil semua, misal Met => MET
- Untuk Aoc, name nya AOC aja

File: modules/General/Models/Schedule.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = ['code', 'category_code' ,'department_code','name','start_date','end_date','description','is_active'];
+
+    /**
+     * Get Schedule category for schedule
+     */
+    public function scheduleCategory()
```
Comment (muhammadcahya) [inline modules/General/Models/Schedule.php:23]: nama relasi langsung category aja,
klo misal table post, sama post category, di relasi model post pake category aja, soalnya waktu manggil model udh pasti ke post category, sama untuk kedepannya

File: modules/General/Models/ScheduleCategory.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = ['code','name','description','is_active'];
+
+    /*
+    * Get schedule for schedule category
+    */
+    public function schedule()
```
Comment (muhammadcahya) [inline modules/General/Models/ScheduleCategory.php:23]: relasi hasMany jamak ya, pake schedules

**Comment by muhammadcahya** [comment]
Please fix it first, then let me know when it's finished

**Comment by mrezav** [comment]
Revision completed, please check again.


Thank you

**Pull request merged**